stl/inc/chrono

```diff
@@ -617,27 +669,1289 @@ namespace chrono {
+    };
+    inline constexpr last_spec last{};
+
+    class day {
+    public:
+        day() = default;
+        constexpr explicit day(unsigned int _Val) noexcept : _Day{static_cast<unsigned char>(_Val)} {}
+
+        constexpr day& operator++() noexcept {
+            ++_Day;
+            return *this;
+        }
+        constexpr day operator++(int) noexcept {
+            return day{_Day++};
+        }
+        constexpr day& operator--() noexcept {
+            --_Day;
+            return *this;
+        }
+        constexpr day operator--(int) noexcept {
+            return day{_Day--};
+        }
+
+        constexpr day& operator+=(const days& _Days) noexcept {
+            _Day += static_cast<unsigned char>(_Days.count());
+            return *this;
+        }
+        constexpr day& operator-=(const days& _Days) noexcept {
+            _Day -= static_cast<unsigned char>(_Days.count());
+            return *this;
+        }
+
+        _NODISCARD constexpr explicit operator unsigned int() const noexcept {
+            return _Day;
+        }
+        _NODISCARD constexpr bool ok() const noexcept {
+            return _Day >= 1 && _Day <= 31;
+        }
+
+    private:
+        unsigned char _Day;
+    };
+
+    _NODISCARD constexpr bool operator==(const day& _Left, const day& _Right) noexcept {
+        return static_cast<unsigned int>(_Left) == static_cast<unsigned int>(_Right);
+    }
+    _NODISCARD constexpr strong_ordering operator<=>(const day& _Left, const day& _Right) noexcept {
+        return static_cast<unsigned int>(_Left) <=> static_cast<unsigned int>(_Right);
+    }
+
+    _NODISCARD constexpr day operator+(const day& _Left, const days& _Right) noexcept {
+        return day{static_cast<unsigned int>(_Left) + _Right.count()};
+    }
+    _NODISCARD constexpr day operator+(const days& _Left, const day& _Right) noexcept {
+        return _Right + _Left;
+    }
+    _NODISCARD constexpr day operator-(const day& _Left, const days& _Right) noexcept {
+        return day{static_cast<unsigned int>(_Left) - _Right.count()};
+    }
+    _NODISCARD constexpr days operator-(const day& _Left, const day& _Right) noexcept {
+        return days{
+            static_cast<int>(static_cast<unsigned int>(_Left)) - static_cast<int>(static_cast<unsigned int>(_Right))};
+    }
+
+    class month {
+    public:
+        month() = default;
+        constexpr explicit month(unsigned int _Val) noexcept : _Month{static_cast<unsigned char>(_Val)} {}
+
+        constexpr month& operator++() noexcept {
+            *this += months{1};
+            return *this;
+        }
+        constexpr month operator++(int) noexcept {
+            month _Temp{*this};
+            ++(*this);
+            return _Temp;
+        }
+        constexpr month& operator--() noexcept {
+            *this -= months{1};
+            return *this;
+        }
+        constexpr month operator--(int) noexcept {
+            const month _Temp{*this};

```

Comment:
@mnatsuhara asked for this `const` originally, but then noticed that it doesn't appear elsewhere. Upon further thought, we are somewhat concerned about affecting the Named Return Value Optimization. In places where it matters (like `vector` etc.) we avoid adding `const` on return values like this, despite our usual preference for `const`, because it will inhibit auto-move-semantics (the special case where lvalues being returned get automatically moved from). While this class doesn't care about move semantics, it seems reasonable to follow the global convention:
```suggestion
            month _Temp{*this};
```